tools/c7n_azure/c7n_azure/resources/storage.py

```diff
@@ -550,3 +550,64 @@ def _process_resource(self, resource):
             resource['name'],
             StorageAccountUpdateParameters(enable_https_traffic_only=self.data.get('value'))
         )
+
+
+@Storage.filter_registry.register('soft-delete')
+class SoftDeleteFilter(Filter):
```

Comment:
let's make this a ValueFilter such that you can do filters like:

```yaml
policies:
- name: require-soft-delete-enabled-with-7-days
  description: require soft-delete enabled and days greater than 7
  resource: azure.storage
  filters:
    - or:
      - type: soft-delete
        key: enabled
        value: false
      - type: soft-delete
        key: days
        value: 7
        op: lt
```

Reply:
This is done, but now I changed the filter to be `blob-services` since its not just about soft delete. This also greatly expands the functionality of the filter.